tools/cldr-code/src/main/java/org/unicode/cldr/util/NameGetter.java

```diff
@@ -0,0 +1,547 @@
+                localeSeparator,
+                null,
+                null);
+    }
+
+    /**
+     * Returns the name of the given bcp47 identifier. Note that extensions must be specified using
+     * the old "\@key=type" syntax.
+     *
+     * @param localeOrTZID the bcp47 identifier (locale or timezone ID)
+     * @param onlyConstructCompound if true, returns "English (United Kingdom)" instead of "British
+     *     English"
+     * @param altPicker Used to select particular alts. For example, SHORT_ALTS can be used to get
+     *     "English (U.K.)" instead of "English (United Kingdom)"
+     * @return the name of the given bcp47 identifier
+     */
+    public synchronized String getNameFromBCP47BoolAlt(
+            String localeOrTZID,
+            boolean onlyConstructCompound,
+            Transform<String, String> altPicker) {
+        return getNameFromBCP47BoolAltPaths(localeOrTZID, onlyConstructCompound, altPicker, null);
+    }
+
+    /**
+     * Returns the name of the given bcp47 identifier. Note that extensions must be specified using
+     * the old "\@key=type" syntax.
+     *
+     * @param localeOrTZID the bcp47 identifier (locale or timezone ID)
+     * @param onlyConstructCompound if true, returns "English (United Kingdom)" instead of "British
+     *     English"
+     * @param altPicker Used to select particular alts. For example, SHORT_ALTS can be used to get
+     *     "English (U.K.)" instead of "English (United Kingdom)"
+     * @return the name of the given bcp47 identifier
+     */
+    public synchronized String getNameFromBCP47BoolAltPaths(
+            String localeOrTZID,
+            boolean onlyConstructCompound,
+            Transform<String, String> altPicker,
+            Set<String> paths) {
+        return getNameFromManyThings(
+                localeOrTZID,
+                onlyConstructCompound,
+                cldrFile.getWinningValueWithBailey(GETNAME_LOCALE_KEY_TYPE_PATTERN_PATH),
+                cldrFile.getWinningValueWithBailey(GETNAME_LOCALE_PATTERN_PATH),
+                cldrFile.getWinningValueWithBailey(GETNAME_LOCALE_SEPARATOR_PATH),
+                altPicker,
+                paths);
+    }
+
+    /**
+     * Returns the name of the given bcp47 identifier. Note that extensions must be specified using
+     * the old "\@key=type" syntax. Only used by ExampleGenerator.
+     *
+     * @param localeOrTZID the bcp47 identifier (locale or timezone ID)
+     * @param onlyConstructCompound if true, returns "English (United Kingdom)" instead of "British
+     *     English"
+     * @param localeKeyTypePattern the pattern used to format key-type pairs
+     * @param localePattern the pattern used to format primary/secondary subtags
+     * @param localeSeparator the list separator for secondary subtags
+     * @param paths if non-null, fillin with contributory paths
+     * @return the name of the given bcp47 identifier
+     */
+    private synchronized String getNameFromManyThings(
```

Comment:
again, a builder model that sets up options might be better than a kitchen sink set of parameters.

`new NameFormatter(locale).onlyConstructCompound(true).withLocalePatter(pattern)…`